thoth/storages/graph/postgres.py

```diff
@@ -4831,54 +4944,69 @@ def sync_security_indicator_aggregated_result(self, document: dict) -> None:
+                )
 
-            si_aggregated_run, _ = SecurityIndicatorAggregatedRun.get_or_create(
-                session,
-                si_aggregated_run_document_id=document_id,
-                datetime=metadata['datetime'],
-                severity_high_confidence_high=result.get('SEVERITY.HIGH__CONFIDENCE.HIGH') or 0,
-                severity_high_confidence_low=result.get('SEVERITY.HIGH__CONFIDENCE.LOW') or 0,
-                severity_high_confidence_medium=result.get('SEVERITY.HIGH__CONFIDENCE.MEDIUM') or 0,
-                severity_high_confidence_undefined=result.get('SEVERITY.HIGH__CONFIDENCE.UNDEFINED') or 0,
-                severity_low_confidence_high=result.get('SEVERITY.LOW__CONFIDENCE.HIGH') or 0,
-                severity_low_confidence_low=result.get('SEVERITY.LOW__CONFIDENCE.LOW') or 0,
-                severity_low_confidence_medium=result.get('SEVERITY.LOW__CONFIDENCE.MEDIUM') or 0,
-                severity_low_confidence_undefined=result.get('SEVERITY.LOW__CONFIDENCE.UNDEFINED') or 0,
-                severity_medium_confidence_high=result.get('SEVERITY.MEDIUM__CONFIDENCE.HIGH') or 0,
-                severity_medium_confidence_low=result.get('SEVERITY.MEDIUM__CONFIDENCE.LOW') or 0,
-                severity_medium_confidence_medium=result.get('SEVERITY.MEDIUM__CONFIDENCE.MEDIUM') or 0,
-                severity_medium_confidence_undefined=result.get('SEVERITY.MEDIUM__CONFIDENCE.UNDEFINED') or 0,
-                number_of_analyzed_files=result['number_of_analyzed_files'],
-                number_of_files_total=result['number_of_files_total'],
-                number_of_files_with_severities=result['number_of_files_with_severities'],
-                number_of_filtered_files=result['number_of_filtered_files'],
-                number_of_python_files=result['Python.nFiles'],
-                number_of_lines_with_comments_in_python_files=result['Python.comment'],
-                number_of_blank_lines_in_python_files=result['Python.blank'],
-                number_of_lines_with_code_in_python_files=result['Python.code'],
-                total_number_of_files=result['SUM.nFiles'],
-                total_number_of_lines=result['SUM.n_lines'],
-                total_number_of_lines_with_comments=result['SUM.comment'],
-                total_number_of_blank_lines=result['SUM.blank'],
-                total_number_of_lines_with_code=result['SUM.code']
-            )
-
-            SIAggregated.get_or_create(
-                session,
-                si_aggregated_run_id=si_aggregated_run.id,
-                python_package_version_entity_id=python_package_version_entity.id,
-            )
+                python_package_version, _ = PythonPackageVersion.get_or_create(
+                    session,
+                    package_name=package_name,
+                    package_version=package_version,
+                    python_package_index_id=python_package_index.id,
```

Comment:
This component don't have to create entries in PPV or PPVE. The method above should return PPVE ID and PPV ID to be passed to:

```
SIAggregated.get_or_create(
                    session,
                    si_aggregated_run_id=si_aggregated_run.id,
                    python_package_version_entity_id=python_package_version_entity.id,
                    python_package_version=python_package_version.id,
                )
```

If the method above does not return those required values it will fail. If some package is not solved for some reason, we cannot sync the corresponding package in the database for consistency. The system should heal itself automatically rescheduling eventually an SI.